python/pyspark/ml/pipeline.py

```diff
@@ -242,7 +242,7 @@ def load(self, path):
             return JavaMLReader(self.cls).load(path)
         else:
             uid, stages = PipelineSharedReadWrite.load(metadata, self.sc, path)
-            return PipelineModel(stages=stages)._resetUid(uid)
+            return self.cls(stages=stages)._resetUid(uid)
```

Comment:
My guess is that `self.cls` was just added as a corresponding private variable `className` in Scala side just to make it easier to match in the future. This variable seems supposed to be a `PipelineModel` class.

Reply:
I do see it now in the Scala side. But does it not then make sense to adapt the Scala side as well. I'll be honest that I am not a Scala developer, so I wouldn't know how to change it correctly without breaking. However, I see no reason to NOT change it to use className/self.cls when those variables are expected to be a PipelineModel type anyways. By changing it, it just adds a bit more flexibility to the way it can be inherited.